### DISCUSSION TITLE
compress the model to facilitate MD exploration in the model_devi stage

### DISCUSSION THREAD

**Discussion by LiangMD-BGI** [deepmodeling/dpgen#559] [Ideas]
Hello,

I found that after training, dpgen only freezes the model but does not compress it.
This is somewhat unsatisfactory as one may want to facilitate molecular dynamics simulations in the stage of model_devi.
Can this possibly be improved?

**Comment by amcadmus** [comment]
We do not see urgent need of model compression for the exploration step, because DPMD is relatively much cheaper than training and first principles calculations.

**Comment by AnguseZhang** [comment]
Thanks for your suggestions!  The reason that we've not  implemented  `dp compress` in `DP-GEN` is that  the number of atoms is usually quite small, e.g,  less than one hundred, and thus `compress` won't bring a significant speed-up.  We would suggest `dp compress` after you've collected all training data and try to perform a large-scale production-mode simulation.

**Comment by chtchelkatchev** [comment]
My systems usually include many atoms (alloys) and include phase transitions. So lammps stage takes a lot of time: several hours sometimes. It would be very helpful if dpgen implemented compression of the model with subsequent cleaning the compressed model after the end of lammps calculations. (By the way, option of cleaning model.ckpt files would be also good in dpgen.)

**Reply by wanghan-iapcm** [reply]
please refer to https://github.com/deepmodeling/dpgen/pull/607 for compression

**Reply by chtchelkatchev** [reply]
Thank You for quick answer. I applied option dp_compress in dpgen. However the procedure fails with multi-task training in dpgen. It would be nice to upgrade "dp compress" in deepmd for multi-task running.

**Reply by iProzd** [reply]
Thank you for utilizing the multi-task training feature in the dpgen process. Multi-task training for now is just for model training process. We appreciate your support and would like to inform you that our current support is preliminary, but we are actively working on enhancing it. For now, you can manually freeze a single branch in your multi-task model using the 'dp freeze' command without '—united-model' option, and then proceed with the usual 'dp compress' operation on the chosen single model. We hope this solution is helpful to you, and please stay tuned for more updates!
